__tests__/app.test.js

```diff
@@ -299,3 +299,58 @@ describe('POST /api/articles/:article_id/comments', () => {
     
 });
 
+describe('PATCH /api/articles/:article_id', () => {
+    it('responds with updated article', () => {
+        const articleId = 1;
+        const incrementVotes = 5;
+        let initialVotes;
+      
+        return request(app)
+          .get(`/api/articles/${articleId}`)
+          .expect(200)
+          .then((res) => {
+            initialVotes = res.body.article.votes;
+            return request(app)
+              .patch(`/api/articles/${articleId}`)
+              .send({ inc_votes: incrementVotes })
+              .expect(200);
+          })
+          .then((res) => {
+            const updatedVotes = res.body.article.votes;
+            expect(updatedVotes).toBe(initialVotes + incrementVotes);
+          });
+    });
+
+    it('responds with status 404 if article_id does not exist', () => {
+        const articleId = 999999;
+        return request(app)
+          .patch(`/api/articles/${articleId}`)
+          .send({ inc_votes: 5 })
+          .expect(404)
+          .then((res) => {
+            const { body } = res;
+            expect(body.msg).toBe('No articles found');
+        });
+    });
+    
+    it('decrements the current article vote property by 100 when given inc_votes of -100', () => {
+        const articleId = 1;
+        const decrementVotes = -100;
+        let initialVotes;
+      
+        return request(app)
+          .get(`/api/articles/${articleId}`)
+          .expect(200)
+          .then((res) => {
+            initialVotes = res.body.article.votes;
+            return request(app)
+              .patch(`/api/articles/${articleId}`)
+              .send({ inc_votes: decrementVotes })
+              .expect(200);
+          })
+          .then((res) => {
+            const updatedVotes = res.body.article.votes;
+            expect(updatedVotes).toBe(initialVotes + decrementVotes);
+          });
+    });
```

Comment:
Just missing a few tests:

- [ ]  400 invalid id
- [ ]  400 inc_vote not an integer in the patch body
- [ ] 200/400 inc_vote key missing - either respond with an error or make no change.

Reply:
i've refactored my tests and added these three